go.mod

```diff
@@ -1,49 +1,53 @@
 module github.com/grafana/athena-datasource
 
-go 1.20
+go 1.21
+
+toolchain go1.21.0
```

Comment:
If I understood correctly, the upgraded grafana-plugin-sdk-go version declares go 1.21 which now defines a minimum build version of go under this keyword `toolchain`. https://go.dev/blog/toolchain

`toolchain` was introduced in go 1.21